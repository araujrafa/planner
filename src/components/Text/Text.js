import React from 'react';

export default ({value, onChangeText, type, label}) => {
  return (
    <div className="form-group">
      <label>{label}</label>
      <input
        className="form-control" 
        type={type}
        value={value}
        onChange={onChangeText}
      />  
    </div>
  )
}