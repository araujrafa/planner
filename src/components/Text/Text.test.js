import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import Text from './Text';

describe('Text component', () => {
  const props = {
    value: '',
    onChangeText: jest.fn(),
    label: 'Quantidade de dias',
    type: 'number'
  }

  test('Text changes the class when hovered', () => {  
    const component = renderer.create(
      <Text {...props}/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Text call funtion after change', () => {
    const wrapper = shallow( <Text {...props}/>);
  
    wrapper.find('input').simulate('change');
  
    expect(props.onChangeText).toHaveBeenCalledTimes(1);
  });
})