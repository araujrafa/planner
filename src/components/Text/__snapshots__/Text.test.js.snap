// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Text changes the class when hovered 1`] = `
<div
  className="form-group"
>
  <label>
    Quantidade de dias
  </label>
  <input
    className="form-control"
    onChange={[Function]}
    type="number"
    value=""
  />
</div>
`;
